chirp/drivers/tdh8.py

```diff
@@ -0,0 +1,2222 @@
+                                                PTTID_VALUES[_mem.pttid]))
+        mem.extra.append(rs)
+
+        # Busylock
+        rs = RadioSetting("bcl", "Busy Lock",
+                          RadioSettingValueList(BCLOCK_VALUES,
+                                                BCLOCK_VALUES[_mem.bcl]))
+        mem.extra.append(rs)
+
+        # Scan add
+        scan_val_list = []
+        for x in range(25):
+            a = self._get_scan(x)
+            for i in range(0, 8):
+                scan_val = (a._generators["scan" + str(i + 1)])
+                # print(str(scan_val)[3])
+                used_scan_val = str(scan_val)[3]
+                scan_val_list.append(used_scan_val)
+
+        rs = RadioSetting(
+            "scanadd", "Scan Add", RadioSettingValueList(
+                SCAN_VALUES, SCAN_VALUES[int(scan_val_list[number - 1])]))
+        mem.extra.append(rs)
+
+        rs = RadioSetting(
+            "freqhop", "Frequency Hop", RadioSettingValueList(
+                FREQHOP_VALUES, FREQHOP_VALUES[_mem.freqhop]))
+        mem.extra.append(rs)
+
+        return mem
+
+    def _set_mem(self, number):
+        return self._memobj.memory[number]
+
+    def _set_nam(self, number):
+        return self._memobj.names[number]
+
+    def _get_scan_list(self, scan_data):
+
+        # scan_val_list - Get all scans Add data 1-200 digits
+        scan_val_list = []
+        for x in range(25):
+            a = self._get_scan(x)
+            for i in range(0, 8):
+                scan_val = (a._generators["scan" + str(i + 1)])
+                used_scan_val = str(scan_val)[3]
+                scan_val_list.append(used_scan_val)
+
+        # used_scan_list - 25 structures, split the scan added
+        # data into 25 groups of 8 bits each
+        used_scan_list = []
+        count_num = 1
+        for i in range(0, len(scan_val_list), 8):
+            used_scan_list.append(scan_val_list[i:i + 8])
+            count_num += 1
+        # Determine whether it is a standard number that can be divisible
+        # Which group is the scan addition located in the modified channel
+        if scan_data % 8 != 0:
+            x_list = scan_data / 8
+            y_list = scan_data % 8
+
+        else:
+            x_list = (scan_data / 8) - 1
+            y_list = scan_data
+
+        return ([x_list, y_list])
+
+    def get_block_xy(self, number):
+        if number % 8 != 0:
+            x_list = number / 8
+            y_list = number % 8
+
+        else:
+            x_list = (number / 8) - 1
+            y_list = number
+
+        return ([x_list, y_list])
+
+    def set_memory(self, mem):
+        _mem = self._get_mem(mem.number)
+        _nam = self._get_nam(mem.number)
+
+        if mem.empty:
+            _mem.set_raw("\xff" * 16)
+            _nam.set_raw("\xff" * 16)
+            return
+
+        _mem.set_raw("\x00" * 16)
+
+        if self._ham and self.radio_ident == "P31185":
```

Comment:
Why and? When would they ever be different? Anyway, as above, this would be better delegated to a helper method in the ham subclass.